src/main/java/wolox/training/models/Book.java

```diff
@@ -65,6 +66,7 @@ public String getTitle() {
     }
 
     public void setTitle(String title) {
+        Preconditions.checkNotNull(title);
```

Comment:
You could also check for the string not being empty.
You  can do `Preconditions.checkArgument(title != null && !title.isEmpty());`
Also, add the check for the setPages setter, so it checks it's > 0.

Reply:
Got it!